src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -218,6 +218,39 @@ public Command drive(InputStream vx, InputStream vy, Supplier<Rotation2d> headin
                         getPose().getRotation().getRadians(), heading.get().getRadians()))));
   }
 
+  public Command goTo(Pose2d targetPose) {
+    // store current position to calculate nessesary components
+    Pose2d initialPose = getPose();
+
+    // calculates components of 3D vector
+    double deltaX = targetPose.getX() - initialPose.getX();
+    double deltaY = targetPose.getY() - initialPose.getY();
+    double deltaTheta =
+        targetPose.getRotation().getRadians() - initialPose.getRotation().getRadians();
+
+    // calulates magninutde of vector
+    // sqrt((x)^2 + (y)^2 + (w*r)^2)
+    double magnitude =
+        Math.sqrt(
+            Math.pow(deltaX, 2)
+                + Math.pow(deltaY, 2)
+                + Math.pow(deltaTheta * DriveToConstants.ANGULAR_RADIUS, 2));
+
+    var pid =
```

Comment:
this can be outside of the method and make sure to set the goal (and make sure it makes sense)